File: frontend/src/types/agentOperations.ts
export type AgentWriteOperation = {
  id: string;
  operation_type: string;
  source_path: string;
  actor: string;
  result_status: string;
  api_error_code: string | null;
  request_summary: Record<string, unknown>;
  diagnostics: Record<string, unknown>;
  related_refs: Record<string, unknown>;
  candidate_id: string | null;
  correction_id: string | null;
  episode_id: string | null;
  canonical_record_ref: string | null;
  bounded_excerpt: string | null;
  created_at: string;
  updated_at: string;
};

export type AgentOperationFilters = {
  actor: string;
  source_path: string;
  operation_type: string;
  result_status: string;
  has_error: string;
  created_from: string;
  created_to: string;
};

File: backend/src/kinlayer_backend/schemas/agent_operations.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from kinlayer_backend.schemas.common import APIModel, ListResponse


class AgentWriteOperationRead(APIModel):
    id: str
    operation_type: str
    source_path: str
    actor: str
    result_status: str
    api_error_code: str | None = None
    request_summary: dict[str, Any]
    diagnostics: dict[str, Any]
    related_refs: dict[str, Any]
    candidate_id: str | None = None
    correction_id: str | None = None
    episode_id: str | None = None
    canonical_record_ref: str | None = None
    bounded_excerpt: str | None = None
    created_at: datetime
    updated_at: datetime


AgentWriteOperationList = ListResponse[AgentWriteOperationRead]
